app/controllers/dashboard/dashboard-activity-controller.js

```diff
@@ -14,41 +14,114 @@ const auth = require('../../services/auth_service.js')
-    fromDateTime = _.get(req, 'query.fromDateTime', null)
-    toDateTime = _.get(req, 'query.toDateTime', null)
+  // Get any custom date ranges
+  const customFomDateTime = _.get(req, 'query.fromDateTime', null)
+  const customToDateTime = _.get(req, 'query.toDateTime', null)
 
-    const validFromDateTime = fromDateTime && moment(fromDateTime).isValid()
-    const validToDateTime = toDateTime && moment(toDateTime).isValid()
-    const getCustomDateRangeDays = (fromDateTime, toDateTime) => moment(toDateTime).diff(moment(fromDateTime), 'days')
+  try {
+    const {fromDateTime, toDateTime} = getTransactionDateRange(period, {
+      fromDateTime: customFomDateTime,
+      toDateTime: customToDateTime
+    })
 
-    if (!validFromDateTime || !validToDateTime || getCustomDateRangeDays(fromDateTime, toDateTime) > 31) {
-      logger.error(`[${correlationId}] Invalid custom date range specified`)
-      res.status(400)
-      response(req, res, 'dashboard/index', {
+    const transactionsPeriodString = `fromDate=${encodeURIComponent(datetime(fromDateTime, 'date'))}&fromTime=${encodeURIComponent(datetime(fromDateTime, 'time'))}&toDate=${encodeURIComponent(datetime(toDateTime, 'date'))}&toTime=${encodeURIComponent(datetime(toDateTime, 'time'))}`
+
+    logger.info(`[${correlationId}] successfully logged in`)
+
+    if (isADirectDebitAccount(gatewayAccountId)) {
+      // todo implement transaction list for direct debit
+      return response(req, res, 'dashboard/index', {
         name: req.user.username,
         serviceId: req.service.externalId,
         activityError: true,
         period
       })
-      return
     }
-    fromDateTime = decodeURIComponent(fromDateTime)
-    toDateTime = decodeURIComponent(toDateTime)
+    connectorClient().getTransactionSummary({
+      gatewayAccountId,
+      correlationId,
+      fromDateTime,
+      toDateTime
+    }, (connectorData, connectorResponse) => {
+      const activityResults = connectorResponse.body
+      response(req, res, 'dashboard/index', {
+        name: req.user.username,
+        serviceId: req.service.externalId,
+        activity: activityResults,
+        successfulTransactionsState: 'payment-success',
+        fromDateTime,
+        toDateTime,
+        period,
+        transactionsPeriodString
+      })
+    })
+      .on('connectorError', (error, connectorResponse) => {
+        const status = _.get(connectorResponse, 'statusCode', 404)
+        logger.error(`[${correlationId}] Calling connector to get transactions summary failed -`, {
+          service: 'connector',
+          method: 'GET',
+          status,
+          error
+        })
+        res.status(status)
+        response(req, res, 'dashboard/index', {
+          name: req.user.username,
+          serviceId: req.service.externalId,
+          activityError: true,
+          period
+        })
+      })
+  } catch (err) {
+    logger.error(`[${correlationId}] ${err.message} -`, {
+      service: 'frontend',
```

Comment:
is the service meant to be selfservice?

Reply:
In this instance, the service we call is connector, so that error message is flagged as relating to connector. There is a general catch block for the controller itself that sends back an error of 'self-service' for internal exceptions.